### ISSUE TITLE
[Refactor] Move methods isWindows()/isLinux()/isMacOSX() outside ResourceTest

### ISSUE THREAD

**Issue by fedejeanne** [eclipse-platform/eclipse.platform#448]
The reason for this issue is mentioned in the PR #446 (see [comment](https://github.com/eclipse-platform/eclipse.platform/pull/446#discussion_r1190047831) and this change should be made only after that PR is merged.

### Option 1: moving to a utility class

These 3 static methods should either be moved to a utility class:
* `org.eclipse.core.tests.resources.ResourceTest.isWindows()`
* `org.eclipse.core.tests.resources.ResourceTest.isLinux()` (still not merged into *master*)
* `org.eclipse.core.tests.resources.ResourceTest.isMacOSX()`  (still not merged into *master*)

### Option 2: move them to the `Platform` class
This would mean: remove these methods and directly call new methods in `Platform`.

## Pre-requisites
PR #446 should be already merged.

**Comment by fedejeanne** [comment]
Thank you @akurtakov, I'll look into that. 

It might be a major change to move to JUnit5 though, so please don't get your hopes too high, he. But I#ll let you know what I find out ;)

**Comment by akurtakov** [comment]
I have made one such migration - it's pretty straightforward though touches many places (making test methods is not mandatory but I did for the sake of testing whether anything will break). See https://github.com/eclipse-platform/eclipse.platform.ui/commit/d179e6ec58ebef3ca376de46cf17dd50aa24e7b5

**Comment by vogella** [comment]
Simple tests migration (import package/ public removal, new asserts) might be possible to migrate with chatgpt. Copy the test into it and ask to migrate to Junit5. One of my clients reported to me that this worked relatively fine for him.

**Comment by akurtakov** [comment]
There is also https://github.com/junit-pioneer/convert-junit4-to-junit5 if anyone is interested .

**Comment by HeikoKlare** [comment]
Option 2 as proposed in https://github.com/eclipse-platform/eclipse.platform/pull/446#discussion_r1190877282 would also be fine. Maybe it would be better to simply have some `Platform.isOS(...)` method to which you can pass the OS constants defined in `Platform` instead of having methods for every platform to ease adaption for further platforms.
However, since new (relevant) platforms do not come up that often it might also be okay to simply have the three methods for the major platforms within the `Platform` class.

**Comment by fedejeanne** [comment]
Quick update (I finally got time to get back to this):
@akurtakov I tried the tool you suggested but it didn't yield any changes. I suppose it's because the code is not JUnit4 but JUnit3 . Have you tried the tool before? Did it work for you?

@vogella I tried with ChatGPT too and it's not very promising: I asked it to translate this JUnit3 test to JUnit5...

```
import junit.framework.TestCase;

public class MyJUnit3Test extends TestCase {
	
	public void testAssertFalse() {
		assertFalse(false);
	}
	
	public void testAssertTrue() {
		assertTrue("Kaboom!",false);
	}
}
```

And this is what came back:

```

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class MyJUnit5Test {

	@Test
	public void assertFalseTest() {
		assertFalse(false);
	}

	@Test
	public void assertTrueTest() {
		assertFalse("Kaboom!", false); // compilation error
	}
}
```

The logic in `assertTrueTest` changes (`assertFalse` instead of `assertTrue`) and also `assertFalse(String, boolean)` is not JUnit5 (the parameters are inverted) --> compilation error.

:-\

**Comment by vogella** [comment]
@fedejeanne you can train ChatGPT to do it better, I just tested on in the Bing engine (which uses GPT 4.0). First answer was as yours, so I instructed it:

````
JUnit 5 also inverts the parameter in assertFalse, can you adjust your answer accordingly?
````

Answer from Bing:

````
You are correct! I apologize for the mistake. Here is the updated code:

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class MyJUnit5Test {
    @Test
    public void testAssertFalse() {
        assertFalse(false);
    }

    @Test
    public void testAssertTrue() {
        assertTrue(false, "Kaboom!");
    }
}

Thank you for bringing this to my attention. Let me know if you have any other questions.
````

**Comment by fedejeanne** [comment]
Sorry for the late reply.

I tried it with ChatGPT but it didn't work:
* ChatGPT does not translate the whole class but only the first 2 or 3 test methods. I tried asking for the whole class several times and it doesn't work, it keeps saying "sure, here it is" and gives me the same result as before.
* It doesn't preserve the hierarchy of the class, which causes compilation errors since the inherited members get lost.

I tried doing it manually but it's a very expensive thing, one needs to:
1. Manually break the hierarchy to avoid extending from `TestCase` (JUnit3)
2. Migrate the test classes
3. Migrate the test suites

As I mentioned before (https://github.com/eclipse-platform/eclipse.platform/issues/448#issuecomment-1543802486), breaking the hierarchy is the most expensive part since even the classes at the bottom of the hierarchy tree use several fields and methods from their superclass(es) which means one has to break the hierarchy "from above".

For example, if I were to migrate Tests that extend `CoreTest`, I would need to break one of these:
* `CoreTest` extends`TestCase`: 7211 compile errors
* `FileSystemTest` extends`CoreTest`: 331 compile errors
* `ResourceTest` extends`CoreTest`: 11164 compile errors
* `RuntimeTest` extends`CoreTest`: 991 compile errors

And the best case scenario would be that all the subclasses belong to the same test suite otherwise a suite would have mixed tests (JUnit3 and JUnit5).

All in all: a full blown migration is doable but there isn't much to gain and I don't think it justifies the effort.

**Comment by HeikoKlare** [comment]
This has been addressed via:
- #704
